SearchAPI/CMR/SubQuery.py

```diff
@@ -115,18 +115,18 @@ def get_results(self):
         logging.debug('Processing page 1')
 
         session = self.asf_session()
-        response = self.get_page(session, 1)
+        response = self.get_page(session)
 
         if response is None:
             return
 
         self.hits = int(response.headers['CMR-hits'])
-        self.sid = None
-        if self.scroll:
-            self.sid = response.headers['CMR-Scroll-Id']
-            request.cmr_scroll_sessions.append(self.sid)
+        self.cmr_search_after = None
+
+        if self.scroll and 'CMR-Search-After' in response.headers:            
```

Comment:
I think we can get rid of self.scroll entirely. The purposes of that flag was to try to guess whether a scroll session would be required at all, and only use it when necessary, in order to reduce active scroll sessions before they are even initiated. If I'm reading the docs correctly, the search-after response header is always available, so we can just blindly make use of it. Makes it simpler!